### DISCUSSION TITLE
oneSide Link with 2 different sides

### DISCUSSION THREAD

**Discussion by TobiasRama** [clientIO/joint#1854] [Q&A] [answered]
Is it possible to create a oneSide link with 2 different sides? I would like to have a fixed source anchor in the middle of the bottom of the shape and a fixed anchor on the top of the target.

**Comment by kumilingus** [comment, marked as answer]
For that, you need to write your own router: https://jsfiddle.net/kumilingus/tbf9vun0/

**Reply by TobiasRama** [reply]
I tried to write my own router like you did, but this only works for shapes that already exist. In my example the user should be able to spawn shapes by clicking a Button and to connect them via drag and drop which is realized with the Connect() element tool. Is there a way to define this router in my use case? I think it doesn't work, because the router doesn't know the target yet, when the user creates a link.

**Reply by kumilingus** [reply]
You need to check if both `target` and `source` elements exist.

https://jsfiddle.net/kumilingus/tbf9vun0/

**Reply by TobiasRama** [reply]
Thanks for your help! One Last question: How can I access the position of the mouse while dragging the link? I do not want to draw a straight line if the target doesn't exist, so I need the position of the mouse to calculate my own line.

**Reply by kumilingus** [reply]
I've left a comment in the previous [jsfiddle](https://jsfiddle.net/kumilingus/tbf9vun0/) with an alternative route while dragging.

This is it: https://jsfiddle.net/kumilingus/rpo2b7aw/

If the link is not connected to an element the `source` resp. `target` is a point (e.g. `{ x: 10, y: 20 }`).
